src/key/helper.js

```diff
@@ -334,24 +335,38 @@ export function sanitizeKeyOptions(options, subkeyDefaults = {}) {
 
   options.sign = options.sign || false;
 
-  if (options.curve) {
+  if (!util.isString(options.algorithm)) {
     try {
-      options.curve = enums.write(enums.curve, options.curve);
+      options.algorithm = enums.read(enums.publicKey, options.algorithm);
     } catch (e) {
-      throw new Error('Not valid curve.');
+      throw new Error('Invalid key algorithm');
     }
-    if (options.curve === enums.curve.ed25519 || options.curve === enums.curve.curve25519) {
-      options.curve = options.sign ? enums.curve.ed25519 : enums.curve.curve25519;
-    }
-    if (options.sign) {
-      options.algorithm = options.curve === enums.curve.ed25519 ? enums.publicKey.eddsa : enums.publicKey.ecdsa;
-    } else {
-      options.algorithm = enums.publicKey.ecdh;
-    }
-  } else if (options.rsaBits) {
-    options.algorithm = enums.publicKey.rsaEncryptSign;
-  } else {
-    throw new Error('Unrecognized key type');
+  }
+  switch (options.algorithm) {
+    case "ecc":
+    case "eddsa":
+    case "ecdsa":
+    case "ecdh":
+      try {
+        options.curve = enums.write(enums.curve, options.curve);
+      } catch (e) {
+        throw new Error('Invalid curve');
+      }
+      if (options.curve === enums.curve.ed25519 || options.curve === enums.curve.curve25519) {
+        options.curve = options.sign ? enums.curve.ed25519 : enums.curve.curve25519;
+      }
+      if (options.sign) {
+        options.algorithm = options.curve === enums.curve.ed25519 ? enums.publicKey.eddsa : enums.publicKey.ecdsa;
+      } else {
+        options.algorithm = enums.publicKey.ecdh;
+      }
+      break;
+    case "rsa":
+    case "rsaEncryptSign":
+      options.algorithm = enums.publicKey.rsaEncryptSign;
+      break;
+    default:
+      throw new Error(`Unrecognized key algorithm ${options.algorithm}`);
```

Comment:
```suggestion
      throw new Error(`Unsupported key algorithm ${options.algorithm}`);
```
In case they try to add a subkey to a DSA key or so, it's not necessarily unrecognized (although it could be indeed)